src/ucp/rndv/rndv.c

```diff
@@ -721,23 +724,70 @@ static void ucp_rndv_req_init_zcopy_lane_map(ucp_request_t *rndv_req,
         }
     }
 
-    if (ucs_popcount(lane_map) > 1) {
-        /* remove lanes if bandwidth is too less compare to best lane */
-        ucs_for_each_bit(lane_idx, lane_map) {
+    if (ucs_popcount(lane_map) <= 1) {
+        goto out;
+    }
+
+    /* Remove lanes if bandwidth is too small compared to the best lane */
+    ucs_for_each_bit(lane_idx, lane_map) {
+        ucs_assert(lane_idx < UCP_MAX_LANES);
+        lane       = lanes[lane_idx];
+        rsc_index  = ep_config->key.lanes[lane].rsc_index;
+        iface_attr = ucp_worker_iface_get_attr(ep->worker, rsc_index);
+        lane_bw    = ucp_tl_iface_bandwidth(context, &iface_attr->bandwidth);
+
+        if ((lane_bw / max_lane_bw) <
```

Comment:
If lane indexes are ordered according to their bandwidth, why do we go further after facing first low-bw lane?

Reply:
ok, it's possible to mask out all following lanes using `lane_map & = UCS_MASK(lane_idx)` instead of iterating, but we have to fill `rndv_req->send.rndv.rkey_index` with `UCP_NULL_RESOURCE` anyway.
Also please note that the number of lanes is supposed to be quite low number (currently up to 4)

Reply:
if I'm not mistaken, lanes are not sorted by their BW characteristics

Reply:
they are taken from rma_bw_lanes, which are sorted